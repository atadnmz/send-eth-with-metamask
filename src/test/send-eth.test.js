import React from "react";
import { shallow, mount } from "enzyme";
import SendEthContainer from "../containers/send-eth/index";
import SendEthComponent from "../components/send-eth/index";

describe("SendEthContainer", () => {
  test('should render correctly in "debug" mode', () => {
    const component = shallow(<SendEthContainer debug />);

    expect(component).toMatchSnapshot();
  });
});

describe("render props and function calls", () => {
  let component;
  let etherAmount = 1;
  let userAddress = "0x123";
  let acountBalance = 100;
  let loading = false;
  let changeEtherAmount = jest.fn();
  let changeUserAddress = jest.fn();
  let onSend = jest.fn();
  beforeAll(() => {
    component = shallow(
      <SendEthComponent
        etherAmount={etherAmount}
        userAddress={userAddress}
        acountBalance={acountBalance}
        changeEtherAmount={changeEtherAmount}
        changeUserAddress={changeUserAddress}
        onSend={onSend}
        loading={loading}
      />
    );
  });

  test("renders with the passed props", () => {
    expect(component.html()).toMatchSnapshot();
  });
  test("renders ether amount", () => {
    expect(etherAmount).toEqual(
      component
        .find(".send-eth-component_sub-content2-ether input")
        .prop("value")
    );
  });

  test("renders account balance", () => {
    expect(acountBalance.toString()).toEqual(
      component
        .find(".send-eth-component_sub-content1-account-balance p")
        .at(1)
        .text()
    );
  });

  test("renders userAddress", () => {
    expect(userAddress.toString()).toEqual(
      component
        .find(".send-eth-component_sub-content1-user-address input")
        .prop("value")
    );
  });

  test("loading false", () => {
    expect(component.find(".fas fa-spinner fa-pulse").exists()).toBeFalsy();
  });

  test("calls the onSend prop function when clicked footer", () => {
    expect(onSend).not.toHaveBeenCalled();
    component.find("footer").simulate("click");
    expect(onSend).toHaveBeenCalled();
  });

  test("calls changeEtherAmount func", () => {
    const event = { target: { value: "23" } };
    expect(changeEtherAmount).not.toHaveBeenCalled();
    component
      .find(".send-eth-component_sub-content2-ether input")
      .prop("onChange")(event);
    expect(changeEtherAmount).toHaveBeenCalled();
  });

  test("calls changeUserAddress func", () => {
    const event = { target: { value: "23" } };
    expect(changeUserAddress).not.toHaveBeenCalled();
    component
      .find(".send-eth-component_sub-content1-user-address input")
      .simulate("change", event);
    expect(changeUserAddress).toHaveBeenCalled();
  });
});

describe("SendEthContainer mount", () => {
  test("state updated with value of changeEtherAmount box upon change", () => {
    const changeEtherAmount = jest.fn();
    React.useState = jest.fn(() => ["", changeEtherAmount]);
    const component = mount(<SendEthContainer />);
    const event = { target: { value: "10" } };
    component
      .find(".send-eth-component_sub-content2-ether input")
      .simulate("change", event);

    expect(changeEtherAmount).toHaveBeenCalledWith("10");
  });
  test("state updated with value of userAddress box upon change", () => {
    const setUserAddress = jest.fn();
    React.useState = jest.fn(() => ["", setUserAddress]);
    const component = mount(<SendEthContainer />);
    const event = { target: { value: "11" } };
    component
      .find(".send-eth-component_sub-content1-user-address input")
      .simulate("change", event);
    expect(setUserAddress).toHaveBeenCalledWith("11");
  });
});
