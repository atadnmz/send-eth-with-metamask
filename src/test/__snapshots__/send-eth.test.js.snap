// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SendEthContainer should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;

exports[`render props and function calls renders with the passed props 1`] = `"<div class=\\"send-eth-component\\"><header class=\\"send-eth-component_header\\"><span class=\\"send-eth-component_icon\\"><i class=\\"fa fa-bars\\" style=\\"color:white;cursor:pointer\\"></i></span></header><div class=\\"send-eth-component_main-content\\"><div class=\\"send-eth-component_sub-content1\\"><div class=\\"send-eth-component_sub-content1-user-address\\"><label for=\\"userAddress\\">User Address:</label><input type=\\"text\\" id=\\"userAddress\\" style=\\"border:0px;text-overflow:ellipsis;width:50%;font-family:inherit;font-size:inherit\\" value=\\"0x123\\"/></div><div class=\\"send-eth-component_sub-content1-account-balance\\"><p>Account Balance: </p><p>100</p></div><a class=\\"send-eth-component_sub-content1-link\\" href=\\"https://rinkeby.etherscan.io/address/0x123\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">View account on Etherscan</a></div><div class=\\"send-eth-component_sub-content2\\"><div class=\\"send-eth-component_sub-content2-send-to-friend\\"><p>Send to friend</p></div><div class=\\"send-eth-component_sub-content2-ether\\"><input type=\\"text\\" id=\\"ether\\" value=\\"1\\" style=\\"width:NaNch\\"/><label for=\\"ether\\">Ether</label></div></div></div><footer class=\\"send-eth-component_footer\\"><span>Send</span></footer></div>"`;
